### ISSUE TITLE
Why `extends Node`?

### ISSUE THREAD

**Issue by dueddel** [xsellier/godot-uuid#13]
This is more like a question rather than an actual issue. Is there any particular reason the script extends `Node`? Correct me if I'm wrong please, but wouldn't it be enough to just not extend anything at all (which implicitly extends `Object`)?

It's for sure not a performance killer or something. Though I think, where we don't need any node features it makes more sense to just not extend it, right?

Long story short, it might be no problem to simply delete line `3` from the script.

By the way, thanks for the implementation. I am considering using it for a tool I'd like to build in combination with an SQLite database. 😘

**Comment by xsellier** [comment]
You can add a node to a scene tree but not an object.
So for those who wants to use it as a global instance (singleton) they can.

**Comment by dueddel** [comment]
Okay, yes. I just also played around with it a bit (before seeing your reply). It totally works fine when omitting that `extends Node` line from the script if I (pre)load it like explained in the README:

```gdscript
const uuid_util = preload('res://uuid.gd')
# everything fine in case of uuid.gd doesn't extend Node
```

But when loading the script as a singleton via the Autoload feature in the project settings (as you just said) then it doesn't work. In that case it only does work when extending it from `Node`. Makes totally sense.

And I can understand that people prefer to add the script to the autoloaded classes. So, everything fine. Thank you for your reply.

Let me close that issue. Thanks again. ❤️
